language/e2e-tests/src/tests/verify_txn.rs

```diff
@@ -442,6 +443,26 @@ pub fn test_publish_from_libra_root() {
     );
 }
 
+#[test]
+fn verify_chain_id() {
+    let mut executor = FakeExecutor::from_genesis_file();
+    let sender = AccountData::new(900_000, 0);
+    executor.add_account_data(&sender);
+    let private_key = Ed25519PrivateKey::generate_for_testing();
+    let txn = transaction_test_helpers::get_test_txn_with_chain_id(
+        *sender.address(),
+        0,
+        &private_key,
+        private_key.public_key(),
+        // all tests use TESTING != 253 for chain_id
+        ChainId::new(253),
```

Comment:
Nit: might be slightly fragile. Could you do something like `ChainId::new(ChainId::test() + 1)`?